select-branch-workspace/action.yml

```diff
@@ -0,0 +1,33 @@
+name: Sanitize Branch Name for Terraform Workspace
+description: Removes everything before the final "/" from the branch name and limits the characters.
+inputs:
+  branch-name:
+    description: The name of the branch to be sanitized
+    required: true
+  working-directory:
+    description: the directory the step should be run from
+    required: true
+outputs:
+  workspace-name:
+    description: "Branch name after the final forward slash"
+    value: ${{ steps.sanitize_branch.outputs.sanitized_branch_name }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Sanitize input branch
+      id: sanitize_branch
+      shell: bash
+      run: |
+        pwd
+        cd ${{ inputs.working-directory }}
+        pwd
+        terraform workspace list
```

Comment:
these lines were added when I was troubleshooting to figure out I needed to explicitly set the working directory within the composite step because it did not retain the directory of the workflow that called it. I can remove these but they are inexpensive so I left them for now.